File: 0162-find-peak-element/0162-find-peak-element.rs
impl Solution {
    pub fn find_peak_element(nums: Vec<i32>) -> i32 {
        
        fn search(nums: &[i32], l: usize, r: usize) -> usize
        {
            if l == r
            {
                return l;
            }
            let m : usize = l + (r - l) / 2;
            if nums[m] > nums[m + 1]
            {
                return search(nums, l, m);
            }
            return search(nums, m + 1, r);
        }
        
        return search(&nums, 0, nums.len() - 1) as i32;
    }
}

File: 0162-find-peak-element/0162-find-peak-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findPeakElement = function(nums)
{
    return search(nums, 0, nums.length - 1);
};

var search = function(nums, l, r)
{
    if (l == r)
        return l;
    var mid = l + Math.floor((r - l) / 2);
    if(nums[mid] > nums[mid + 1])
        return search(nums, l, mid);
    return search(nums, mid + 1, r);
}; 